### ISSUE TITLE
Order confirmation page showing sideContent

### ISSUE THREAD

**Issue by mrSingh007** [SAP/cloud-commerce-spartacus-storefront#3938]
Hi,
on order confirmation page side slot is rendered even its not present in response

![image](https://user-images.githubusercontent.com/16721438/62524434-e8cc6e80-b835-11e9-9b93-f0056e101ffb.png)
![image](https://user-images.githubusercontent.com/16721438/62526103-e28bc180-b838-11e9-81a8-3b94cc2cb187.png)

## Environment Details

- Spartacus: 
    "@spartacus/core": "~1.0.1",
    "@spartacus/styles": "~1.0.1",
    "@spartacus/storefront": "~1.0.1",
    "@spartacus/assets": "~1.0.1"

My app module lookes like this:
```
@NgModule({
  imports: [
    B2cStorefrontModule.withConfig({
        ....
    }),
   CustomPaymentMethodModule,  //just overriding component
    CustomReviewOrderModule, // just overriding component
    BrowserModule,
    ...devImports,
    StoreModule.forFeature('myState', reducers),
    EffectsModule.forFeature(effects),
  ],
  declarations: [AppComponent],
  entryComponents: [],
  providers: [{
    provide: CheckoutDetailsService,
    useClass: CustomCheckoutDeatilsService  
 }
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
```
## Expected Results
should not show order button (sideContent)

## Offer to Fix
NO

**Comment by mrSingh007** [comment]
b2c-layout-config.ts have SideContent
 ```https://github.com/SAP/cloud-commerce-spartacus-storefront/blob/develop/projects/storefrontlib/src/recipes/config/b2c-layout-config.ts
    }```

**Comment by marlass** [comment]
This config is not relevant here. It only makes sure the slot exists (slot would be empty). I could reproduce it. Ticket on hold until we find real cause.

**Comment by tobi-or-not-tobi** [comment]
@marlass I believe the config is actually relevant. The `PageLayoutComponent` uses the `PageLayoutService` to resolve the slots from the LayoutConfiguration (`resolveSlots()`).

I've done a quick test with the layout configuration to underline my thinking: 
- I've added a custom layout config for the `ContentPage1Template` which is used to render the sale page (among others).
- I've verified that the configuration overrides the standard slot configuration (the suggestion I gave to @mrSingh007) and takes precedence at runtime.
- I've verified that the configuration indeed causes the DOM (comment) node to be rendered, despite the fact that it is not part of the content.

For this test, I configured the following after the import of the `B2cStorefrontModule`:

```
ConfigModule.withConfig({
  layoutSlots: {
    ContentPage1Template: {
      slots: ['UnknownSlot', 'Section2A', 'Section2B'],
    },
  },
} as LayoutConfig),
```

This is the "expected" result:
![image](https://user-images.githubusercontent.com/4745807/62607459-665bb180-b8fe-11e9-933a-a96ddb4c7cbf.png)

I believe we have 2 issues to solve here:
1. bug (small and easy to fix): drop the SideContent slot if it's indeed not part of the sample data we use (please double check on this, there might be other reasons that the slot is not exposed). 
2. Improvement: consider refactor the code, so that only configured **ánd** actual slots are rendered. This would improve the DOM, however there's a small change that we destroy the layout in case customers would have added specific styling to empty slots. Super small change, so I'd consider doing this anyway. 

I'd suggest to fix 1 with this ticket.

**Comment by marlass** [comment]
@tobi-or-not-tobi I am just wondering why this slot wasn't empty, but filled with components from the previous page?

**Comment by tobi-or-not-tobi** [comment]
That sounds like a serious bug. I actually vaguely remember we've seen this before, but thought it was fixed. The intend is that the PageLayout is rebuild on page visit, so that the former page and slots are rendered from scratch. If the behaviour you describe is true, I'm afraid the former PageSlots are not cleaned up from the store. 

@dunqan can you provide insights here?

**Comment by dunqan** [comment]
Yes, we had this (or similar) bug before, but it was fixed as far as I remember. If it's back, I agree it's serious, so if we it;s confirmed, we should prioritize and have a fix for next patch release if possible.

**Comment by tobi-or-not-tobi** [comment]
@marlass thanks for reporting. 

@mrSingh007 hope you got unblocked by the conclusions we made in the ticket. Thanks for reporting.

**Comment by dunqan** [comment]
QA steps: 
1. Open sale pages
2. Confirm that if there is no Section2B slot in the DOM

**Comment by Matejk00** [comment]
Checked, there is no Section2B in a DOM.
